src/App.tsx

```diff
@@ -1,26 +1,153 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { getTodos, USER_ID, deleteTodo, addTodo } from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoItem } from './components/TodoItem';
+import { Todo } from './types/Todo';
+import { Filters } from './types/Filtres';
+import { ErrorNotification } from './components/ErrorNotification';
+import { filterTodos } from './utils/filterTodos';
+import { ErrorType } from './types/ErrorType';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(ErrorType.Empty);
+  const [currentFilter, setCurrentFilter] = useState<Filters>(Filters.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const onAddTodo = async (todoTitle: string) => {
+    setTempTodo({
+      id: 0,
+      title: todoTitle,
+      completed: false,
+      userId: USER_ID,
+    });
+    try {
+      const newTodo = await addTodo({ title: todoTitle, completed: false });
+
+      setTodos(prev => [...prev, newTodo]);
+    } catch (err) {
+      setErrorMessage(ErrorType.AddTodo);
+      throw err;
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const onDeleteTodo = async (todoId: number) => {
+    setLoadingTodoIds(prev => [...prev, todoId]);
+    try {
+      await deleteTodo(todoId);
+
+      setTodos(prev => prev.filter(todo => todo.id !== todoId));
+    } catch (err) {
+      setErrorMessage(ErrorType.DeleteTodo);
+      throw err;
```

Comment:
You should not throw an error here, because you don't use `try/catch` in place where you use this function